readme.md

```diff
@@ -99,13 +99,18 @@ console.log("IO is a side effect!");
 
 ## Idempotency
 
-> A function is said to be idempotent if it has no side-effects on multiple
-executions with the the same input parameters.
+> A function is said to be idempotent if there are no additional effects of calling it a second time with the same parameters.
```

Comment:
I don't think that's really it.  That doesn't sound very different from "pure".  The point is that if the output of an idempotent is fed back into that same function, it returns an equal value.  It's the notion that there are sinks into which values fall, never to arise again.

This is not really a question of reference equality, although such can be applied to it if you somehow take into account the unchanging state of an object/record used as a parameter.  `Absolute value` and `sort` are classic examples of functions, but there is a very similar notion for procedures, and `delete record`, even though it has important side-effects, is still idempotent.